dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
> Beside on that, how do you think of using const variable instead enum for INVALID_INDEX?

const variable is definitely better. I had my head in C land (rather than C++ land) when I made InvalidIndex.

> DynamicsTypes.h is just my preference rather than create different type defining files like InvalidIndex.h if it doesn't cause too much coupling. But I'm fine with Ptr.h and InvalidIndex.h.

I would normally agree, but I feel like the existing ``dynamics.h`` serves the role of a ``DynamicsTypes.h``. I made a separate header for InvalidIndex just because I couldn't think of anything else to put in there. If we find ourselves with more simple types belonging to the dynamics namespace, I'd be okay with consolidating them into a single header.